proposals/0304-structured-concurrency.md

```diff
@@ -1018,7 +1019,7 @@ extension ThrowingTaskGroup: AsyncSequence {
 
   /// Wait for a task to complete and return the result or thrown error packaged in
   /// a `Result` instance. Returns `nil` only when there are no tasks left in the group.
-  mutating func nextResult() async -> Result<ChildTaskResult, Error>?
+  mutating func nextResult() async throws -> Result<ChildTaskResult, Error>?
```

Comment:
This is wrong. Thanks for helping spot this though, will add a test covering it.

We either return nil, success or failure, there is no room for throwing here; the signature in the proposal is correct.